vissl/data/torchvision_dataset.py

```diff
@@ -0,0 +1,88 @@
+from typing import Tuple, List
+
+from PIL import Image
+from fvcore.common.file_io import PathManager
+from torch.utils.data import Dataset
+from torchvision.datasets import CIFAR10, CIFAR100, STL10, MNIST
+
+from vissl.utils.hydra_config import AttrDict
+
+
+class TorchvisionDatasetName:
+    """
+    Names of the Torchvision datasets currently supported in VISSL.
+    """
+    CIFAR10 = "CIFAR10"
+    CIFAR100 = "CIFAR100"
+    STL10 = "STL10"
+    MNIST = "MNIST"
+
+
+class TorchvisionDataset(Dataset):
+    """
+    Adapter dataset to available datasets in Torchvision.
+    The selected dataset is based on the name provided as argument.
+    This name must match the names of the dataset in torchvision.
+
+    Args:
+        cfg (AttrDict): configuration defined by user
+        data_source (string): data source ("torchvision_dataset") [not used]
+        path (string): path to the dataset
+        split (string): specify split for the dataset (either "train" or "val").
+        dataset_name (string): name of dataset (should be one of TorchvisionDatasetName).
+    """
+
+    def __init__(self,
+                 cfg: AttrDict,
+                 data_source: str,
+                 path: str,
+                 split: str,
+                 dataset_name: str):
+        super().__init__()
+        assert PathManager.isdir(path), f"Directory {path} does not exist"
+        self.dataset_name = dataset_name
+        self.path = path
+        self.split = split.lower()
+        self.dataset = self._load_dataset()
+
+    def _load_dataset(self):
+        is_train_split = self.split == "train"
+        if self.dataset_name == TorchvisionDatasetName.CIFAR10:
+            return CIFAR10(self.path, train=is_train_split)
+        elif self.dataset_name == TorchvisionDatasetName.CIFAR100:
+            return CIFAR100(self.path, train=is_train_split)
+        elif self.dataset_name == TorchvisionDatasetName.STL10:
+            stl_split = "train" if is_train_split else "test"
+            return STL10(self.path, split=stl_split)
+        elif self.dataset_name == TorchvisionDatasetName.MNIST:
+            return MNIST(root=self.path, train=is_train_split)
+        else:
+            raise ValueError(f"Unsupported dataset {self.dataset_name: str}")
```

Comment:
nit: if we use the enum, we can remove the if/else need :) but it doesn't need to be addressed in this PR :)

Reply:
The problem is that the `self.dataset_name` is a string and not an enumeration. And since it is plugged down as follow, I cannot really change the fact that the input is a string:

```
DATASET_SOURCE_MAP = {
    "disk_filelist": DiskImageDataset,
    "disk_folder": DiskImageDataset,
    "torchvision_dataset": TorchvisionDataset,
    "synthetic": SyntheticImageDataset,
}
```

And the string itself is given through the configuration, so it might be anything (and not necessarily a member of `TorchvisionDatasetName`). So in short, I think we need this default case and the associated input validation.